vaadin-messages-flow-parent/vaadin-messages-flow/src/main/java/com/vaadin/flow/component/messages/MessageInput.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.messages;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.DomEvent;
+import com.vaadin.flow.component.EventData;
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.dependency.NpmPackage;
+import com.vaadin.flow.shared.Registration;
+
+/**
+ * Server-side component for the {@code vaadin-message-input} element. The
+ * component displays a text-area to input a message and a button to send it.
+ *
+ * @author Vaadin Ltd.
+ */
+@Tag("vaadin-message-input")
+@JsModule("@vaadin/vaadin-messages/src/vaadin-message-input.js")
+@NpmPackage(value = "@vaadin/vaadin-messages", version = "v2.0.0-alpha1")
+public class MessageInput extends Component {
+
+    /**
+     * The {@code submit} event which is fired by the component.
+     */
+    @DomEvent("submit")
+    public static class SubmitEvent extends ComponentEvent<MessageInput> {
+
+        private final String value;
+
+        /**
+         * Creates the event.
+         *
+         * @param source     the source component
+         * @param fromClient true if the event comes from the client
+         * @param value      the value of the input
+         */
+        public SubmitEvent(MessageInput source, boolean fromClient,
+                @EventData("event.detail.value") String value) {
+            super(source, fromClient);
+            this.value = value;
+        }
+
+        /**
+         * Gets the value of the input.
+         *
+         * @return the value of the input
+         */
+        public String getValue() {
+            return value;
+        }
+    }
+
+    /**
+     * Gets the current content of the input field.
+     *
+     * @return the value of the input field
+     */
+    public String getValue() {
+        return getElement().getProperty("value");
+    }
+
+    /**
+     * Sets the value on the input field.
+     *
+     * @param value the value to set on the input field
+     */
+    public void setValue(String value) {
+        getElement().setProperty("value", value);
+    }
+
+    /**
+     * Adds a listener that is called when the user submits the value of the
+     * input field, which can be obtained with {@link SubmitEvent#getValue()}.
```

Comment:
Could explain that it's fired when clicking the send button or pressing the enter key

Reply:
The discussion in Slack is leaning towards the option of not having `getValue`/`setValue` at all at this point.